diff --git a/elements/core-deps/systemd-base.bst b/elements/core-deps/systemd-base.bst
--- a/elements/core-deps/systemd-base.bst
+++ b/elements/core-deps/systemd-base.bst
@@ -10,7 +10,7 @@ sources:
   track: v*
   exclude:
   - '*-rc*'
-  ref: v260.2-0-gf1d0952a125b96b7ab2f1ff29a87448ade8ac29b
+  ref: v261-0-gde9dbc30cbea730329870bf6be2aab91dac98d38
 
 build-depends:
 - freedesktop-sdk.bst:public-stacks/runtime-minimal.bst
@@ -95,6 +95,7 @@ variables:
     -Ddns-servers="%{fallback-dns-servers}"
     -Ddefault-dnssec=no
     -Didn=true
+    -Dsysinstall=true
     -Dman=enabled
     -Dhtml=enabled
     -Dtpm=true
